File: projects/fizzbuzz/python/fizz_buzz.py


def fizz_buzz(num):
    string = ' '
    for i in range(1,num+1):
        if (i%3==0 and i%5==0 and i % 7==0):
            string += "fizzbuzzbazz"
            #print("fizzbuzzbazz")
        elif (i%3==0 and i%5==0):
            string += "fizzbuzz"
            #print("fizzbuzz")
        elif (i%3==0 and i%7==0):
            string += "fizzbazz"
            #print("fizzbazz")
        elif (i%5==0 and i%7==0):
            string += "buzzbazz"
            #print("buzzbazz")
        elif (i%7==0):
            string += "bazz"
            #print("bazz")
        elif i%3==0:
            string += "fizz"
            #print("fizz")
        elif i%5==0:
            string += "buzz"
            #print("buzz")
        else:
            string += str(i);
            #print(i)
        string += "\n"
    return string;

def main():
    print("Type in a value or use negative one to quit:")
    i = int(input())
    print("Your value is " + str(i))
    if (i > 0 and i < 100):
        n = fizz_buzz(i)
        print(n)
    else:
        print("Quit")


main()        

#if __name__ == '__main__':
    #i = int(input("Type in a value or use negative one to quit:"))
    #if (i > 0 and i < 100):
        #n = fizz_buzz(i)
        #print(n)
    #else:
        #print("Quit")

File: projects/fizzbuzz/java/jonathansorem/FizzBuzz.java
package fizzbuzz;

import java.util.Scanner;

public class FizzBuzz
{
    public static void main(String[] args)
    {
        Scanner input = new Scanner(System.in);
        
        //User enters a number
        System.out.println("Enter a number:");
        int amount = input.nextInt();
        
        //Function call to print
        
        //Print the numbers
        for (int i = 1; i <= amount; i++)
        {
            System.out.println (Print(i));
        }
    }
    public static String Print(int amount)
    {
        //If not divisible by 3 or 5
         if (amount%3 != 0 && amount%5 != 0 && amount%7 != 0)
        {
            return (Integer.toString(amount));
        }
        else
        {
            //If number is divisible by 3, 5, and 7
            if (amount%3 == 0 && amount%5 == 0 && amount%7 == 0)
            {
                return ("FizzBazzBuzz");
            }
            //If number is divisible by 3 and 5
            if (amount%3 == 0 && amount%5 == 0 && amount%7 != 0)
            {
                return ("FizzBazz");
            }
            //If number is divisible by 3 and 7
            if (amount%3 == 0 && amount%5 != 0 && amount%7 == 0)
            {
                return ("FizzBuzz");
            }
            //If number is divisible by 5 and 7
            if (amount%3 != 0 && amount%5 == 0 && amount%7 == 0)
            {
                return ("BazzBuzz");
            }
            //If number is divisible by 3
            if (amount%3 == 0 && amount%5 != 0 && amount%7 != 0)
            {
                return ("Fizz");
            }
            //If number is divisible by 5
            if (amount%5 == 0)
            {
                return ("Buzz");
            }
            //If number is divisible by 7
            if (amount%7 == 0)
            {
                return ("Bazz");
            }
        }   
        return "error";
    }
}
